eng/pipelines/templates/stages/platform-matrix.json

```diff
@@ -41,10 +41,32 @@
           "CmakeArgs": " -DBUILD_TRANSPORT_CURL=ON"
         },
         "Win32Api_release_curl": {
-          "CmakeArgs": " -DBUILD_TESTING=ON -DBUILD_PERFORMANCE_TESTS=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_TRANSPORT_CURL=ON -DMSVC_USE_STATIC_CRT=ON",
+          "CmakeArgs": " -DBUILD_TESTING=ON -DDISABLE_PROXY_TESTS=ON -DBUILD_PERFORMANCE_TESTS=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_TRANSPORT_CURL=ON -DMSVC_USE_STATIC_CRT=ON ",
           "BuildArgs": "--parallel 8 --config Release",
           "PublishMapFiles": "true"
+        }
+      },
+      "TargetArchitecture": {
```

Comment:
These changes introduce consistency in the naming of the first dimension of the matrixes. The platform matrix builds the combinatoric expansion of all of the entries within each element of the array. If the name of the element is empty, it is ignored, otherwise it is used when naming the expansion of the matrix. By providing a name for StaticConfigs, it then applies that name to each of the elements in the matrix. That means that each TargetArchitecture/TargetPlatform element doesn't need to include the name for the StaticConfigs. THe names essentially become the same, but instead of the name for TargetPlatform being `Win_Win32Api_debug_tests`, the `Win` portion of the name comes from the StaticConfigs.

There are other cleanups we can/should make IMHO but this one was cheap and straightforward. And it became really important when I added Win2022 as a StaticConfig.